catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java

```diff
@@ -132,6 +124,26 @@
     @Option(name = "--skip-signature-verification", required = false, multiValued = false, description = "Produces the export zip but does NOT sign the resulting zip file. This file will not be able to be verified on import for integrity and security.")
     boolean unsafe = false;
 
+    private MetacardTransformer transformer;
+
+    private Filter revisionFilter;
+
+    private JarSigner jarSigner = new JarSigner();
+
+    @Reference
+    private StorageProvider storageProvider;
+
+    /**
+     * Generates stateful predicate to filter distinct elements by a certain key in the object.
+     *
+     * @param keyExtractor Function to pull the desired key out of the object
+     * @return the stateful predicate
+     */
+    private static <T> Predicate<T> distinctByKey(Function<? super T, ?> keyExtractor) {
+        Map<Object, Boolean> seen = new ConcurrentHashMap<>();
```

Comment:
❗️ This is going to create a new `ConcurrentHashMap` on each invocation. If the purpose of the map is to ensure uniqueness of the keys, this won't work.

I see that this code existed before; it is tech debt.

You might consider changing the `contentItemsToExport` collection in `doContentExport` to be a Set and provide a custom comparator to assert equality on these keys.

More simply, you can change this method to be:
```java
private static <T> Predicate<T> distinctByKey(Map<Object, Boolean, seen, Function<? super T, ?> keyExtractor) {
    return t -> seen.putIfAbsent(keyExtractor.apply(t), Boolean .TRUE) == null;
}
```
And then create the ConcurrentHashMap right above your stream method and include it in the params passed in.